Cargo.toml

```diff
@@ -49,6 +49,10 @@ c = ["cc"]
 # which use inline assembly and fall back to pure Rust versions (if avalible).
 no-asm = []
 
+# Workaround for codegen backends which haven't yet implemented `f16` and
+# `f128` support. Disabled any intrinsics which use those types.
+no-f16-f128 = []
```

Comment:
Features should usually be additive and not take things away -- so, can't we have an "f16-f128" feature that *enables* these intrinsics, and have them off by default?

Reply:
I think that was just following the `no-asm` feature above, but they could probably both be flipped to be positive

Reply:
This "adds" support for codegen backends that don't support f16/f128. If a `f16-f128` feature is used instead, liballoc would likely enable it when depending on compiler-builtins, which makes it impossible to disable the f16/f128 intrinsics using `--features` when invoking cargo, instead requiring the standard library sources to be patched.

Reply:
... what?^^ I am confused.

We are [currently](https://github.com/rust-lang/rust/pull/125016) (once that PR lands) setting this feature in `std` because of said codegen backends. But that means building `core` alone will still fail with these codegen backends.

> liballoc would likely enable it when depending on compiler-builtins

That sounds wrong to me. Instead, bootstrap should enable the feature when it deems them supported by the environment.

Reply:
> We are https://github.com/rust-lang/rust/pull/125016 (once that PR lands) setting this feature in std because of said codegen backends.

The build system of these codegen backends should set it, not libstd. For example for no-asm, this is what cg_clif does when building the standard library:

https://github.com/rust-lang/rustc_codegen_cranelift/blob/6db27529ddbb6d94e0c990e527aa8341c3d85a79/build_system/build_sysroot.rs#L279

Reply:
I can accept the answer "this is a hack, compiler_builtins is special and has a single consumer and so we can get away with negative features".